src/Common/OpenTelemetryTraceContext.cpp

```diff
@@ -306,6 +302,18 @@ TracingContextHolder::~TracingContextHolder()
         auto shared_span_log = current_thread_trace_context.span_log.lock();
         if (shared_span_log)
         {
+            try
```

Comment:
About the ctor (l. 228ff.): Please see https://github.com/ClickHouse/ClickHouse/pull/39010#issuecomment-1238555649 about other possible exception sources. E.g. `time_since_epoch()` isn't noexcept as per standard. I know, it's a bit silly ... Maybe put a big bold try/catch around the ctor?

Reply:
For this statement `std::chrono::system_clock::now().time_since_epoch()`, `now()` is noexcept, `time_since_epoch` is not declared as noexcept but it's constexpr.

But I think you're right that current ctor is exception safe based on what we do. If it changes, it's easy to lead to errors.